huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go

```diff
@@ -0,0 +1,308 @@
+package ims
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/cbr/v3/backups"
+	"github.com/chnsz/golangsdk/openstack/ims/v2/cloudimages"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+// @API IMS POST /v1/cloudimages/wholeimages/action
+// @API IMS GET /v1/{project_id}/jobs/{job_id}
+// @API IMS GET /v2/cloudimages
+// @API IMS GET /v2/{project_id}/images/{image_id}/tags
+// @API CBR GET /v3/{project_id}/backups/{backup_id}
+// @API IMS PATCH /v2/cloudimages/{image_id}
+// @API IMS POST /v2/{project_id}/images/{image_id}/tags/action
+// @API IMS DELETE /v2/images/{image_id}
+func ResourceEcsWholeImage() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEcsWholeImageCreate,
+		ReadContext:   resourceEcsWholeImageRead,
+		UpdateContext: resourceEcsWholeImageUpdate,
+		DeleteContext: resourceWholeImageDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
```

Comment:
The delete timeout value is 10 minutes in document, please keep the values ​​consistent for both.

Reply:
fixed.